From 6e3de01c0a6f07f82d514d0d7f546a4eb408a445 Mon Sep 17 00:00:00 2001
From: Viktor Ashirov <vashirov@redhat.com>
Date: Mon, 11 Jul 2016 10:10:42 +0200
Subject: [PATCH 3/3] Ticket #48918 - Upgrade to 389-ds-base >= 1.3.5.5 doesn't
 install 389-ds-base-snmp

Bug description:
During upgrade from 389-ds-base version <1.3.5.5 additional
package 389-ds-base-snmp is not installed.

Fix description:
Move "Obsoletes:" section from %description to %package.

https://fedorahosted.org/389/ticket/48918

Reviewed by: nhosoi@redhat.com.

(cherry picked from commit f593ae7790e3372c6812bfe59e58e6d709ec171f)
---
 rpm/389-ds-base.spec.in | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/rpm/389-ds-base.spec.in b/rpm/389-ds-base.spec.in
index d08d379..0924cb5 100644
--- a/rpm/389-ds-base.spec.in
+++ b/rpm/389-ds-base.spec.in
@@ -47,6 +47,8 @@ Group:            System Environment/Daemons
 BuildRoot:        %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 Obsoletes:        %{name}-selinux
 Conflicts:        selinux-policy-base < 3.9.8
+# upgrade path from monolithic %{name} (including -libs & -devel) to %{name} + %{name}-snmp
+Obsoletes:        %{name} <= 1.3.5.4
 Requires:         %{name}-libs = %{version}-%{release}
 Provides:         ldif2ldbm 
 
@@ -152,9 +154,6 @@ isn't what you want. Please contact support immediately.
 Please see http://seclists.org/oss-sec/2016/q1/363 for more information.
 %endif
 
-# upgrade path from monolithic %{name} (including -libs & -devel) to %{name} + %{name}-snmp
-Obsoletes:        %{name} <= 1.3.5.4
-
 %package          libs
 Summary:          Core libraries for 389 Directory Server (%{variant})
 Group:            System Environment/Daemons
@@ -213,13 +212,12 @@ Development Libraries and headers for the 389 Directory Server base package.
 Summary:          SNMP Agent for 389 Directory Server
 Group:            System Environment/Daemons
 Requires:         %{name} = %{version}-%{release}
+# upgrade path from monolithic %{name} (including -libs & -devel) to %{name} + %{name}-snmp
+Obsoletes:        %{name} <= 1.3.5.4
 
 %description      snmp
 SNMP Agent for the 389 Directory Server base package.
 
-# upgrade path from monolithic %{name} (including -libs & -devel) to %{name} + %{name}-snmp
-Obsoletes:        %{name} <= 1.3.5.4
-
 %package          tests
 Summary:          The lib389 Continuous Integration Tests
 Group:            Development/Libraries
-- 
2.4.11

